### PR TITLE
Feature/19 queue schedule

### PR DESCRIPTION
**Pull request by mabiede**
- extract worker scheduler
- improve lwt ignore handling
- add schedule table for "last_run" persistency
- add UI displaying schedules (label, interval, status, last_run)
- improve consistency (schedule, queue worker, matcher)

### CODE REVIEW

File: pool/database/migration/migration_schedule.ml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+let create_schedule_table =
+  Sihl.Database.Migration.create_step
+    ~label:"create registered schedule table"
+    {sql|
+      CREATE TABLE IF NOT EXISTS pool_schedules (
```
Comment (timohuber) [inline pool/database/migration/migration_schedule.ml:5]: Shouldn't there be a reference to the tenant? (`database_label`)

File: pool/web/handler/admin_settings_schedule.ml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+module HttpUtils = Http_utils
+module Message = HttpUtils.Message
+module SmtpAuth = Pool_tenant.SmtpAuth
+
+let active_navigation = "/admin/settings/schedules"
+
+let show req =
+  let result context =
+    Schedule.find_all ()
+    ||> Page.Admin.Settings.Schedule.index context
```
Comment (timohuber) [inline pool/web/handler/admin_settings_schedule.ml:13]: This query should filter by tenant.

File: pool/schedule/entity.ml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  | Every of ScheduledTimeSpan.t
+  | At of ScheduledTime.t
+[@@deriving eq, show]
+
+type t =
+  { label : Label.t
+  ; scheduled_time : scheduled_time
+  ; status : Status.t
+  ; last_run : LastRun.t option
+  ; fcn : unit -> unit Lwt.t [@opaque] [@equal fun _ _ -> true]
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/schedule/entity.ml:69]: No, only when actually executing them. The `@equal` should handle the `fcn` part of it.

File: pool/schedule/entity.ml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  }
+[@@deriving eq, show]
+
+let create label scheduled_time fcn =
+  { label; scheduled_time; status = Status.init; last_run = None; fcn }
+;;
+
+let run_in ?(now = false) schedule =
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/schedule/entity.ml:77]: I removed it completely

File: pool/database/migration/migration_schedule.ml (hunk 1 above)
Comment (mabiede) [reply pool/database/migration/migration_schedule.ml:5]: Maybe in a later version. Currently there are only cross tenant processes.

File: pool/web/handler/admin_settings_schedule.ml (hunk 2 above)
Comment (mabiede) [reply pool/web/handler/admin_settings_schedule.ml:13]: see above

File: pool/web/view/page/page_admin_settings_queue.ml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    ; Field.MaxTries, max_tries |> CCInt.to_string |> txt
+    ; Field.Status, status |> Sihl.Contract.Queue.show_instance_status |> txt
+    ; ( Field.LastErrorAt
+      , last_error_at |> CCOption.map_or ~default formatted_date_time |> txt )
+    ; Field.LastError, last_error |> CCOption.value ~default |> txt
+    ; Field.NextRunAt, next_run_at |> formatted_date_time |> txt
+    ]
+    |> vertical_table
+  in
+  div ~a:[ a_class [ "stack-lg" ] ] [ job_detail ]
```
Comment (timohuber) [inline pool/web/view/page/page_admin_settings_queue.ml:78]:
```suggestion
  div [ job_detail ]
```

This class does not have any effect. You can also remove the div element.

**Pull request merged**